superset/assets/src/visualizations/core/models/ChartPlugin.js

```diff
@@ -0,0 +1,43 @@
+import Plugin from './Plugin';
+import isRequired from '../../../utils/isRequired';
+import { getInstance as getChartMetadataRegistry } from '../registries/ChartMetadataRegistry';
+import { getInstance as getChartComponentRegistry } from '../registries/ChartComponentLoaderRegistry';
+import { getInstance as getChartTransformPropsRegistry } from '../registries/ChartTransformPropsLoaderRegistry';
+
+const IDENTITY = x => x;
+
+export default class ChartPlugin extends Plugin {
+  constructor({
+    metadata = isRequired('metadata'),
+
+    // use transformProps for immediate value
+    transformProps = IDENTITY,
+    // use loadTransformProps for dynamic import (lazy-loading)
+    loadTransformProps,
+
+    // use Chart for immediate value
+    Chart,
+    // use loadChart for dynamic import (lazy-loading)
+    loadChart,
+  } = {}) {
+    super();
+    this.metadata = metadata;
+    this.loadTransformProps = loadTransformProps || (() => transformProps);
+
+    if (loadChart) {
+      this.loadChart = loadChart;
+    } else if (Chart) {
+      this.loadChart = () => Chart;
+    } else {
+      throw new Error('Chart or loadChart is required');
+    }
+  }
+
+  install(key = isRequired('key')) {
```

Comment:
curious what your thoughts are on calling this `register` instead of `install`. I was initially wondering what the difference between `install` and `register` was, and since this just calls `register` for the different registry's, it seems like it could also make sense to just call this method `register` as well?

Reply:
sounds good to me.